src/app/(app)/missing-shoes/Components/MissingShoesFormBuilder.tsx

```diff
@@ -0,0 +1,261 @@
+"use client";
+
+import React from "react";
+import FormWrapper from "@/common/FormWrapper";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { FormTitle } from "@/common/FormTitle/FormTitle";
+import { Input } from "@/components/ui/input";
+import { useMask } from "@react-input/mask";
+import { DateTimePicker } from "@/common/DateTimePicker/DateTimePicker";
+import { CNIC_VALIDATOR_REGEX, PHONENUMBER_VALIDATOR_REGEX } from "@/lib/regex";
+import { ShoesTyes } from "@/constant/constant";
+
+export const missingShoesSchema = z.object({
+  color: z.string().min(1, {
+    message: "Color must be selected.",
+  }),
+  type: z.enum([
+    ShoesTyes.Boots,
+    ShoesTyes.DressShoes,
+    ShoesTyes.Sandals,
+    ShoesTyes.Sneakers,
+    ShoesTyes.Formal,
+    ShoesTyes.Other,
+  ]),
+  size: z.string().refine(
+    (val) => {
+      const sizeNumber = parseFloat(val);
+      return !isNaN(sizeNumber) && sizeNumber >= 1 && sizeNumber <= 50;
+    },
+    {
+      message: "Size must be a number between 1 and 50.",
+    }
+  ),
+  time: z.date().refine((val) => val <= new Date(), {
+    message: "The time cannot be in the future.",
+  }),
+  ownerName: z.string().min(1, {
+    message: "Owner's name is required.",
+  }),
+  ownerPhoneNumber: z
+    .string({ message: "Phone number is required" })
+    .regex(PHONENUMBER_VALIDATOR_REGEX, "Phone number is not valid"),
+  ownerCnic: z.string().regex(CNIC_VALIDATOR_REGEX, "CNIC is not valid"),
+});
+
+export type MissingSchemaType = z.infer<typeof missingShoesSchema>;
+
+export const MissingShoesFormBuilder = () => {
+  const phoneNumberMask = useMask({
+    mask: "___________",
+    replacement: { _: /\d/ },
+  });
+  const cnicMask = useMask({
+    mask: "3____-_______-_",
+    replacement: { _: /\d/ },
+  });
+
+  const form = useForm<MissingSchemaType>({
+    resolver: zodResolver(missingShoesSchema),
+    defaultValues: {
+      color: "",
+      type: ShoesTyes.Formal,
+      size: "",
+      time: undefined,
+      ownerName: "",
+      ownerPhoneNumber: "",
+      ownerCnic: "",
+    },
+    mode: "all",
+  });
+
+  const SHOES_TYPES = [
```

Comment:
can we write this array outside the component because during rendering this array will also render